File: java/src/main/java/com/newnancity/newnanmanager/services/TokenService.java
package com.newnancity.newnanmanager.services;

import com.newnancity.newnanmanager.exceptions.NewNanManagerException;
import com.newnancity.newnanmanager.generated.api.TokenServiceApi;
import com.newnancity.newnanmanager.generated.model.ApiToken;
import com.newnancity.newnanmanager.generated.model.CreateApiTokenRequest;
import com.newnancity.newnanmanager.generated.model.CreateApiTokenResponse;
import com.newnancity.newnanmanager.generated.model.ListApiTokensResponse;
import com.newnancity.newnanmanager.generated.model.UpdateApiTokenRequest;

/** API token management service. */
public final class TokenService extends AbstractService {
    private final TokenServiceApi api;

    public TokenService(TokenServiceApi api) {
        this.api = api;
    }

    public CreateApiTokenResponse createApiToken(CreateApiTokenRequest request) throws NewNanManagerException {
        return execute(() -> api.createApiToken(request).execute());
    }

    public ApiToken getApiToken(int id) throws NewNanManagerException {
        return execute(() -> api.getApiToken(id).execute());
    }

    public ApiToken updateApiToken(int id, UpdateApiTokenRequest request) throws NewNanManagerException {
        return execute(() -> api.updateApiToken(id, request).execute());
    }

    public void deleteApiToken(int id) throws NewNanManagerException {
        executeVoid(() -> api.deleteApiToken(id).execute());
    }

    public ListApiTokensResponse listApiTokens() throws NewNanManagerException {
        return listApiTokens(null, null);
    }

    public ListApiTokensResponse listApiTokens(Integer page, Integer pageSize) throws NewNanManagerException {
        return execute(() -> api.listApiTokens().page(page).pageSize(pageSize).execute());
    }

    public TokenServiceApi raw() {
        return api;
    }
}

File: kotlin/src/main/kotlin/com/nanmanager/bukkit/services/TokenService.kt
package com.nanmanager.bukkit.services

import com.nanmanager.bukkit.http.HttpClient
import com.nanmanager.bukkit.models.*

/** API Token 管理服务；传输和模型解析由 OpenAPI 生成客户端负责。 */
class TokenService(httpClient: HttpClient) : BaseService(httpClient) {
    private val api get() = httpClient.generatedApis.tokens

    fun createApiToken(request: CreateApiTokenRequest): CreateApiTokenResponse {
        val body = toGenerated<com.newnanmanager.generated.models.CreateApiTokenRequest>(request)
        return generated<CreateApiTokenResponse, com.newnanmanager.generated.models.CreateApiTokenResponse> {
            api.createApiTokenWithHttpInfo(body)
        }
    }

    fun getApiToken(request: GetApiTokenRequest): ApiToken {
        return generated<ApiToken, com.newnanmanager.generated.models.ApiToken> {
            api.getApiTokenWithHttpInfo(request.id)
        }
    }

    fun updateApiToken(request: UpdateApiTokenRequest): ApiToken {
        val body = toGenerated<com.newnanmanager.generated.models.UpdateApiTokenRequest>(request)
        return generated<ApiToken, com.newnanmanager.generated.models.ApiToken> {
            api.updateApiTokenWithHttpInfo(request.id, body)
        }
    }

    fun deleteApiToken(request: DeleteApiTokenRequest) {
        generatedVoid { api.deleteApiTokenWithHttpInfo(request.id) }
    }

    fun listApiTokens(request: ListApiTokensRequest): ListApiTokensResponse {
        return generated<ListApiTokensResponse, com.newnanmanager.generated.models.ListApiTokensResponse> {
            api.listApiTokensWithHttpInfo(request.page, request.pageSize)
        }
    }
}
